universal-login-sdk/lib/core/services/WalletService.ts

```diff
@@ -22,11 +29,19 @@ type WalletState = {
   wallet: DeployedWallet;
 };
 
+export type SerializedWalletState = {
+  kind: 'Future';
+  wallet: CounterfactualWallet;
+} | {
+  kind: 'Deployed';
+  wallet: ApplicationWallet;
+};
```

Comment:
That looks like duplication topped with some extra name mixing.

Reply:
> Careful with naming. Let's be consistent here. What is the difference between CounterfactualWallet and FutureWallet? Perhaps you meant SerializableFutureWallet?

Good idea should we also consider renaming `ApplicationWallet` to `SerializableDeployedWallet`?

> We should probably extract class WalletSerializer that converts WalletState to SerializableWalletState, with two methods:

The `WalletService.loadFromStorage` method will soon contain side effects in addition to deserialising. This doesn't fit well with serializer pattern.

Reply:
> That could be split into a number of files:

Separated the models to a separate file